app/components/edit/EditAddress.jsx

```diff
@@ -1,129 +1,252 @@
 
+const EditAddresses = ({ addresses, setAddresses }) => {
+  const [modalState, setModalState] = useState({ type: 'closed' });
 
-const EditAddresses = ({ addresses, setHasLocation, setAddresses }) => (
-  <li key="address" className="edit--section--list--item">
-    <HasPhysicalLocationToggle
-      hasLocation={addresses.length !== 0}
-      setHasLocation={setHasLocation}
-    />
-    <EditAddressCollection collection={addresses} handleChange={setAddresses} />
-  </li>
-);
+  const closeModal = () => setModalState({ type: 'closed' });
+
+  const modalIsOpen = modalState.type !== 'closed';
+  let modalDefaultData;
+  let modalOnSave;
+  switch (modalState.type) {
+    case 'closed':
+      modalDefaultData = {};
+      modalOnSave = () => {};
+      break;
+    case 'add': {
+      modalDefaultData = {};
+      modalOnSave = newData => setAddresses([...addresses, newData]);
+      break;
+    }
+    case 'edit': {
+      modalDefaultData = addresses[modalState.editingIndex];
+      modalOnSave = newData => {
+        const addressesCopy = addresses.slice();
+        addressesCopy[modalState.editingIndex] = { ...newData, dirty: true };
+        setAddresses(addressesCopy);
+      };
+      break;
+    }
+    default:
+      throw new Error(`Unexpected modal state: ${modalState}`);
+  }
+
+  const removeAddress = index => {
+    const address = addresses[index];
+    const newAddresses = addresses.slice();
+    if (address.id) {
+      newAddresses[index] = { ...address, isRemoved: true };
+    } else {
+      newAddresses.splice(index, 1);
+    }
+    setAddresses(newAddresses);
+  };
+
+  return (
+    <li key="address" className="edit--section--list--item">
+      <EditAddressModal
+        isOpen={modalIsOpen}
+        onRequestClose={closeModal}
+        defaultData={modalDefaultData}
+        onSave={modalOnSave}
+      />
+
+      <div className={s.addressListTitle}>Location</div>
+      <div className={s.addressList}>
+        {addresses
+          .map((address, arrayIndex) => [address, arrayIndex])
+          .filter(([address]) => !address.isRemoved)
+          // The displayIndex should skip over any elements that have been
+          // marked for removal, but the arrayIndex must be the actual index of
+          // the element in the addresses array
+          .map(([address, arrayIndex], displayIndexMinusOne) => (
+            <AddressListItem
+              key={address.id || JSON.stringify(address)}
+              displayIndex={displayIndexMinusOne + 1}
+              address={address}
+              onEdit={() => setModalState({ type: 'edit', editingIndex: arrayIndex })}
+              onRemove={() => removeAddress(arrayIndex)}
+            />
+          ))}
+      </div>
+
+      <button
+        className={s.newAddressButton}
+        type="button"
+        onClick={() => setModalState({ type: 'add' })}
+      >
+        Add location
```

Comment:
nit: "Add Location" - other buttons are title case

Reply:
I had been following the mockup designs, which pretty consistently used sentence case, but you're right that this looks weird with some of the other stuff on the page being title case. I'll change this over, and if design prefers sentence case, we can take a pass through everything on the edit page to change that.